app/services/create_cluster_stats_service.rb

```diff
@@ -36,17 +34,27 @@ def call
 
   private
 
+  def network_stat
+    @network_stat ||= ClusterStat.find_or_create_by(network: @network)
+  end
```

Comment:
I can see a lot of changed not related to this story. Please avoid it in the future, especially if the story is big enough without additional refactoring. It makes it harder to find where you made a mistake.